### PR TITLE
Extended declarations to store their packing.

### PR DESCRIPTION
**Pull request by ktopouzi**

### CODE REVIEW

File: src/Parser/ASTConverter.cs [hunk 1]
```diff
@@ -941,6 +941,8 @@ void VisitDeclaration(Declaration decl, AST.Declaration _decl)
             _decl.IsIncomplete = decl.IsIncomplete;
             _decl.IsDependent = decl.IsDependent;
             _decl.DefinitionOrder = decl.DefinitionOrder;
+           _decl.MaxFieldAlignment = decl.MaxFieldAlignment;
```
Comment (tritao) [inline src/Parser/ASTConverter.cs]: Indentation seems off here.

File: src/CppParser/Parser.cpp [hunk 2]
```diff
@@ -3828,6 +3812,19 @@ Declaration* Parser::WalkDeclaration(const clang::Decl* D,
...
     } };
 
+    if (D->hasAttrs())
+    {
+        for (auto it = D->attr_begin(); it != D->attr_end(); ++it)
+        {
+            Attr* Attr = (*it);
+            if (Decl && Attr->getKind() == clang::attr::Kind::MaxFieldAlignment)
+            {
+            auto MFA = cast<clang::MaxFieldAlignmentAttr>(Attr);
```
Comment (tritao) [inline src/CppParser/Parser.cpp]: Indentation seems off here.

File: tests/Common/Common.h [hunk 3]
```diff
@@ -7,6 +7,24 @@
 #include <string>
 #include <vector>
 
+#pragma pack(4)
+class TestPacking
+{
+public:
+	int integer;
```
Comment (ddobrev) [inline tests/Common/Common.h]: 4 spaces instead of tab.

File: tests/Common/Common.h [hunk 4]
```diff
@@ -7,6 +7,24 @@
...
+	bool boolean;
+	double dble;
+	int i2;
+};
+
+#pragma pack(2)
+struct TestP {
```
Comment (ddobrev) [inline tests/Common/Common.h]: { on a new line, also 4 spaces instead of tab.

File: src/CppParser/Parser.cpp [hunk 5]
```diff
@@ -3828,6 +3812,19 @@ Declaration* Parser::WalkDeclaration(const clang::Decl* D,
         break;
     } };
 
+    if (D->hasAttrs())
```
Comment (ddobrev) [inline src/CppParser/Parser.cpp]: Move the check for Decl here. If Decl is null, the entire loop is useless.

File: src/CppParser/Bindings/CLI/AST.cpp [hunk 6]
```diff
@@ -2118,6 +2118,16 @@ void CppSharp::Parser::AST::Declaration::Kind::set(CppSharp::Parser::AST::Declar
     ((::CppSharp::CppParser::AST::Declaration*)NativePtr)->kind = (::CppSharp::CppParser::AST::DeclarationKind)value;
 }
 
+int CppSharp::Parser::AST::Declaration::MaxFieldAlignment::get()
```
Comment (ddobrev) [inline src/CppParser/Bindings/CLI/AST.cpp]: Why are there only changes to the C++/CLI parser bindings, what about C#?

**Pull request merged**